### ISSUE TITLE
Optionally pipe output to pager

### ISSUE THREAD

**Issue by casey** [mattgreen/watchexec#88]
Would it be possible to add an option to pipe `watchexec` output to a pager like `less` or `more`? If a command produces a lot of output, piping it to a pager would prevent the first lines from scrolling off the screen unseen.

I originally filed this against [cargo-watch](https://github.com/passcod/cargo-watch/issues/92), but it might be more appropriate to add it to watchexec itself.

Thanks!

**Comment by yury-egorenkov** [comment]
You can use `watchexec` with pipeline to `less` or `more` by just put command with quotes.

```
watchexec './run-app | less'
```

**Comment by passcod** [comment]
This is a semi-popular request across both here and cargo-watch, but has too much complexity to be done in a dedicated fashion anytime soon. General idea would be something akin to how `bat` works in this regard.

**Comment by unphased** [comment]
Hmm. Is this possible to do in any capacity right now? I was thinking about building a simple CLI app today to better automate some "tiny rapidly changing test cases" workflow. 

watchexec drives something that runs a few checks. Conceptually it's really similar to launching a compile followed by unit tests whenever we save our code, but the focus here is around low friction UI for browsing a lot of results, basically hopping across results. A pager like less is a workable choice as I could just have something like `Test Result:` in the output for each result and i'd just put that into the less search field and scan through it with `n`/`N`. But it is currently impossible (not sure why) to run `less` via a pipeline under watchexec right now. If we could, watchexec should work well with `-r --stop-signal=TERM` to drive a `less` whereby the `less` stays keyboard-interactive. Perhaps there is something going on that requires a PTY layer here. I'd like to avoid using a PTY when it doesn't seem necessary though.

I do know I can run some simple non-curses based terminal manipulation under watchexec. But I dont want to build a TUI tool right now. 

This did lead me to another idea, though, which I will use instead. It's to produce output files in a suitable structure for browsing with another UI like `lf`.

**Comment by passcod** [comment]
Yeah the entire problem is who has responsibility (watchexec, less, or the command) to control, update, and interrupt the pager; in my research some years ago this essentially required a custom pager implementation, and no existing crates.io paging library had enough to satisfy requirements.

PTY support in watchexec could help some, and will come eventually.

**Comment by unphased** [comment]
So I'm back on this topic because i realized that it's definitely time to get a responsive `git diff`. This is because previously i'd be the one editing my code. Either i'm interacting with a `less` under a terminal or I'm funneling the output into a neovide split, in either case i'm blasting out two quick keystrokes to refresh the git diff. 

But now that most code is being written by an AI I have a renewed need for the open git diff to auto-update to show the latest version, and the only sane way to do this to be honest is launch `git diff` behind a file watcher. The pager is what's being the spanner in the works here. Ideally the pager is augmented with file watching capability. But not having such deep integration has a measure of unixy appeal if it's doable. 

The way toward this (i'm thinking anyway for now) is to decouple them a bit where we have a script that's just a loop that relinquishes the terminal and launches git diff under less, and watchexec runs separately and uses signals or other ipc to perform the single needed event (kill the pager!). 

It's even better if this looper process also performs the launching of the watchexec as a separate child, directing watchexec's output into a log file instead of the tty. There should be a way to do this cleanly that won't leave orphaned `watchexec` instances around too often.
